feat(components): youtube and iframe (#569)
johnsonandjohnson/Bodiless-JS: packages/bodiless-components/doc/YouTube.md

# YouTube Component

>   One can use this to compose and place a YouTube video on the page, that uses

Reviewer: ```suggestion
  The YouTube Component can be used to place a YouTube video on a page via
```

Author: fixed in 1843998dc9049836238361ae2de68d32d51b289a

>   the bodiless edit system.

Reviewer: ```suggestion
  the BodilessJS edit interface.
```

Author: fixed in bab3dd81c7b4bd2494285dcfdf2f53c0aafba8fe

  ``` js
  import { YouTube } from '@bodiless/components';

  <YouTube nodeKey="youtube" />
  ```

>   One can also use the HOC version of this which can then be apply to other components. But

Reviewer: ```suggestion
You can also use the HOC version of this which can then be applied to other components. But
```

Author: fixed in 1843998

  the underlining component must accept the same props as an `iframe` tag. Simply pass
  the node key to the asBodilessYouTube function and then use the returned HOC

  ``` js
  import { CustomYouTube } from 'my-library';
  import { asBodilessYouTube } from '@bodiless/components';

  const YouTube = asBodilessYouTube('customYouTube')(CustomYouTube);

  <YouTube />
  ```

>   One can configure YouTube player settings leveraging withYouTubePlayerSettings HOC. For example, lets configure AutoPlay for our YouTube component.

Reviewer: ```suggestion
You can configure YouTube player settings leveraging withYouTubePlayerSettings HOC. For example, to configure AutoPlay for the YouTube component:
```

Author: fixed in 1843998

  ``` js
  import { YouTube } from '@bodiless/components';

  const AutoPlayYouTube = withYouTubePlayerSettings({
    autoplay: true,
    mute: true,
  })(YouTube);

  <AutoPlayYouTube nodeKey="youtube" />

>   ```

Author: fixed in 1843998
